geo/src/algorithm/relate/geomgraph/intersection_matrix.rs

```diff
@@ -220,6 +219,48 @@ impl IntersectionMatrix {
             && self.0[CoordPos::Outside][CoordPos::Inside] == Dimensions::Empty
             && self.0[CoordPos::Outside][CoordPos::OnBoundary] == Dimensions::Empty
     }
+
+    /// Directly accesses this matrix
+    ///
+    /// ```
+    /// use geo_types::{LineString, Rect, line_string};
+    /// use geo::algorithm::{coordinate_position::CoordPos, dimensions::Dimensions, relate::Relate};
+    ///
+    /// let line_string: LineString<f64> = line_string![(x: 0.0, y: 0.0), (x: 10.0, y: 0.0), (x: 5.0, y: 5.0)];
+    /// let rect = Rect::new((0.0, 0.0), (5.0, 5.0));
+    ///
+    /// let intersection = line_string.relate(&rect);
+    ///
+    /// // The intersection of the two interiors is empty, because no part of the string is inside the rect
+    /// assert_eq!(intersection.get(CoordPos::Inside, CoordPos::Inside), Dimensions::Empty);
+    ///
+    /// // The intersection of the two string's interior with the rect's boundary is one-dimensoinal, because part of the first line overlaps one of the rect's edges
```

Comment:
```suggestion
    /// // The intersection of the line string's interior with the rect's boundary is one-dimensoinal, because part of the first line overlaps one of the rect's edges
```